src/api-client.js

```diff
@@ -79,20 +79,20 @@ export default class ApiClient {
       return null;
     }
 
-    const { token } = await this.auth.refreshToken(
+    const session = await this.auth.refreshToken(
       this.refreshToken,
       this.refreshBackend,
       this.refreshExpiration,
       true,
     );
 
     if (this.onRefreshToken) {
-      this.onRefreshToken(token);
+      this.onRefreshToken(session.token, session);
```

Comment:
Haven't analyze this one in depth, but couldn't you just feed `session` here? (no reply necessart)

Reply:
It'll break compatibility